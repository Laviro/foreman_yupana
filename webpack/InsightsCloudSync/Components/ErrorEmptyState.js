import React from 'react';
import { useSelector } from 'react-redux';
import {
  EmptyState,
  EmptyStateIcon,
  EmptyStateVariant,
  EmptyStateBody,
} from '@patternfly/react-core';
import { ExclamationCircleIcon } from '@patternfly/react-icons';
import { translate as __, sprintf } from 'foremanReact/common/I18n';
import { selectError } from '../InsightsCloudSyncSelectors';

export const ErrorEmptyState = () => {
  const error = useSelector(selectError);
  return (
    <EmptyState variant={EmptyStateVariant.xl}>
      <EmptyStateIcon icon={ExclamationCircleIcon} />
      <EmptyStateBody>
        <p>
          {sprintf(
            __(
              'Something went wrong while getting the data, the server returned the following error: %s'
            ),
            error
          )}
        </p>
      </EmptyStateBody>
    </EmptyState>
  );
};
