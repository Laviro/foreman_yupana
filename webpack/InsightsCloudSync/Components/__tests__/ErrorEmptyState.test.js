import React from 'react';
import { mount } from '@theforeman/test';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import { ErrorEmptyState } from '../ErrorEmptyState';

const mockStore = configureMockStore();
const store = mockStore({});
describe('ErrorEmptyState', () => {
  it('render', () => {
    const component = mount(
      <Provider store={store}>
        <ErrorEmptyState />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
