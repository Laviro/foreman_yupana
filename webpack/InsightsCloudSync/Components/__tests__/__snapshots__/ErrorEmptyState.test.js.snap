// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorEmptyState render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ErrorEmptyState>
    <EmptyState
      variant="xl"
    >
      <div
        className="pf-c-empty-state pf-m-xl"
      >
        <div
          className="pf-c-empty-state__content"
        >
          <EmptyStateIcon
            icon={[Function]}
          >
            <ExclamationCircleIcon
              aria-hidden="true"
              className="pf-c-empty-state__icon"
              color="currentColor"
              noVerticalAlign={false}
              size="sm"
            >
              <svg
                aria-hidden="true"
                aria-labelledby={null}
                className="pf-c-empty-state__icon"
                fill="currentColor"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                />
              </svg>
            </ExclamationCircleIcon>
          </EmptyStateIcon>
          <EmptyStateBody>
            <div
              className="pf-c-empty-state__body"
            >
              <p>
                Something went wrong while getting the data, the server returned the following error: [object Object]
              </p>
            </div>
          </EmptyStateBody>
        </div>
      </div>
    </EmptyState>
  </ErrorEmptyState>
</Provider>
`;
